From 28726180e25cf7cfcf271000179619098109d447 Mon Sep 17 00:00:00 2001
From: KenC1014 <kenken4016@gmail.com>
Date: Fri, 14 Feb 2025 22:07:08 -0600
Subject: [PATCH 023/150] add o3d import

---
 GEMstack/onboard/perception/fusion.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/GEMstack/onboard/perception/fusion.py b/GEMstack/onboard/perception/fusion.py
index 159fbc68..ed6cd760 100644
--- a/GEMstack/onboard/perception/fusion.py
+++ b/GEMstack/onboard/perception/fusion.py
@@ -2,6 +2,7 @@ from cv_bridge import CvBridge
 from sensor_msgs.msg import Image, PointCloud2
 from ultralytics import YOLO
 from fusion_utils import *
+import open3d as o3d
 import rospy
 import message_filters
 import os
-- 
2.38.1

