From 5aa571b04bb4979a46c5942ea5c648c4f3f83b88 Mon Sep 17 00:00:00 2001
From: KenC1014 <kenken4016@gmail.com>
Date: Sat, 15 Feb 2025 02:30:54 -0600
Subject: [PATCH 030/150] remove empty line

---
 GEMstack/onboard/perception/fusion.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/GEMstack/onboard/perception/fusion.py b/GEMstack/onboard/perception/fusion.py
index 0ac34638..c9f1101d 100644
--- a/GEMstack/onboard/perception/fusion.py
+++ b/GEMstack/onboard/perception/fusion.py
@@ -104,7 +104,6 @@ class Fusion3D():
             ros_extracted_pedestrian_pc2 = create_point_cloud(flattened_pedestrians_3d_pts)
             self.pub_pedestrians_pc2.publish(ros_extracted_pedestrian_pc2)
 
-        
         # Used for visualization
         if(self.visualization):
             ros_img = self.bridge.cv2_to_imgmsg(cv_image, 'bgr8')
-- 
2.38.1

