From 006b5550658e6f0028f65ef59d3b61c1c6f3505b Mon Sep 17 00:00:00 2001
From: KenC1014 <kenken4016@gmail.com>
Date: Sat, 15 Feb 2025 16:42:45 -0600
Subject: [PATCH 033/150] update comments

---
 GEMstack/onboard/perception/fusion.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/GEMstack/onboard/perception/fusion.py b/GEMstack/onboard/perception/fusion.py
index bfd39f7c..6208d5e0 100644
--- a/GEMstack/onboard/perception/fusion.py
+++ b/GEMstack/onboard/perception/fusion.py
@@ -90,11 +90,11 @@ class Fusion3D():
                 if len(extracted_pts) < 1:
                     continue
                 
-                # Apply ground and max distance filter to extracted 5D points
+                # Apply ground and max distance filter to the extracted 5D points
                 extracted_pts = filter_ground_points(extracted_pts, self.ground_threshold)
                 extracted_pts = filter_far_points(extracted_pts)
                 
-                # Extract 2D pedestrians points camera frame
+                # Extract 2D pedestrians points in camera frame
                 extracted_2d_pts = list(extracted_pts[:, :2].astype(int))
                 flattened_pedestrians_2d_pts = flattened_pedestrians_2d_pts + extracted_2d_pts
 
-- 
2.38.1

