From 98f445f6cb914f95db3bea071a47a7de5b071055 Mon Sep 17 00:00:00 2001
From: Lukas Dumasius <80981746+lukasdumasius@users.noreply.github.com>
Date: Wed, 19 Feb 2025 09:51:04 -0600
Subject: [PATCH 061/150] Update pedestrian_detection.py

---
 GEMstack/onboard/perception/pedestrian_detection.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/GEMstack/onboard/perception/pedestrian_detection.py b/GEMstack/onboard/perception/pedestrian_detection.py
index 9970c6a2..58b8e7ac 100644
--- a/GEMstack/onboard/perception/pedestrian_detection.py
+++ b/GEMstack/onboard/perception/pedestrian_detection.py
@@ -263,6 +263,8 @@ class PedestrianDetector2D(Component):
         for ind in range(num_objs):
             obj_center = (None, None, None) if obj_centers[ind].size == 0 else obj_centers[ind]
             obj_dim = (None, None, None) if obj_dims[ind].size == 0 else obj_dims[ind]
+
+            # Part 3 - Change creating this to start frame
             self.current_agents[track_ids[ind]] = (
                 AgentState(
                     track_id = track_ids[ind],
-- 
2.38.1

