From 89d8daf3a4dbedf8b1a9c569b3dd772e303fd110 Mon Sep 17 00:00:00 2001
From: LucasEby <lucaseby@outlook.com>
Date: Mon, 24 Feb 2025 18:24:16 -0500
Subject: [PATCH 086/150] Removed unnecessary print statement

---
 GEMstack/onboard/perception/pedestrian_detection.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/GEMstack/onboard/perception/pedestrian_detection.py b/GEMstack/onboard/perception/pedestrian_detection.py
index 0a0b8d0a..f12cec27 100644
--- a/GEMstack/onboard/perception/pedestrian_detection.py
+++ b/GEMstack/onboard/perception/pedestrian_detection.py
@@ -185,9 +185,6 @@ class PedestrianDetector2D(Component):
         vehicle_agents = self.create_vehicle_dict(agents, backOrder)
         # Create a list containing the ids of the new agents in the vehicle frame agent's order
 
-        print("VEHICLE AGENTS")
-        print(vehicle_agents)
-
         return vehicle_agents
     
     def create_vehicle_dict(self, agents: List[AgentState], backOrder: List[int]):
-- 
2.38.1

