From 5c6af6a43929aa3b7ccd5e6d0f328d8210a1106a Mon Sep 17 00:00:00 2001
From: nvikramraj <nvikramraj1305@gmail.com>
Date: Mon, 24 Feb 2025 19:06:45 -0600
Subject: [PATCH 087/150] edge case fix

---
 GEMstack/onboard/perception/pedestrian_detection.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/GEMstack/onboard/perception/pedestrian_detection.py b/GEMstack/onboard/perception/pedestrian_detection.py
index f12cec27..d32fd32e 100644
--- a/GEMstack/onboard/perception/pedestrian_detection.py
+++ b/GEMstack/onboard/perception/pedestrian_detection.py
@@ -137,6 +137,7 @@ class PedestrianDetector2D(Component):
 
         # Edge cases
 
+        # Initial vehicle pose data
         if(self.current_vehicle_state == None and self.previous_vehicle_state == None):
             self.current_vehicle_state = vehicle
             # We get vehicle state from GNSS in global state
@@ -150,8 +151,13 @@ class PedestrianDetector2D(Component):
             self.previous_vehicle_state = self.current_vehicle_state
             self.current_vehicle_state = vehicle
 
+        # edge case to handle no pedestrian data
         if(self.current_agent_obj_dims == {}):
             return self.current_agents
+
+        # edge case to handle empty lists and missing data
+        if(len(self.current_agent_obj_dims['pose']) == 0 or len(self.current_agent_obj_dims['dims'])) or (len(self.current_agent_obj_dims['pose']) != len(self.current_agent_obj_dims['dims'])):
+            return self.current_agents
         
         # Update current time:
         self.curr_time = vehicle.pose.t
-- 
2.38.1

