From 91784d71db6860df83533cf2c44bbb3c05c8fa29 Mon Sep 17 00:00:00 2001
From: krishauser <hauser.kris@gmail.com>
Date: Thu, 24 Apr 2025 21:22:57 -0400
Subject: [PATCH 126/150] Fixed math error

---
 GEMstack/mathutils/transforms.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/GEMstack/mathutils/transforms.py b/GEMstack/mathutils/transforms.py
index 4c00cbff..7a5a7a8d 100644
--- a/GEMstack/mathutils/transforms.py
+++ b/GEMstack/mathutils/transforms.py
@@ -75,8 +75,8 @@ def point_segment_distance(x,a,b) -> Tuple[float,float]:
     elif udotv > vnorm**2:
         return vector_norm(vector_sub(x,b)),1
     else:
-        param = udotv/vnorm
-        return vector_norm(vector_sub(u,vector_mul(v,param/vnorm))),param
+        param = udotv/vnorm**2
+        return vector_norm(vector_sub(u,vector_mul(v,param))),param
 
 def rotate2d(point, angle : float, origin=None):
     """Rotates a point about the origin by an angle"""
-- 
2.38.1

